pydicom/dataelem.py

```diff
@@ -865,27 +885,133 @@ def DataElement_from_raw(
+    RawDataElement
+        The raw data element
+    dict
+        Any kwargs to pass into downstream handlers.
+    """
+    if raw.tag in _LUT_DESCRIPTOR_TAGS and raw.value:
+        value = kwargs.get('value', [])
         # We only fix the first value as the third value is 8 or 16
         try:
             if value[0] < 0:
                 value[0] += 65536
         except TypeError:
             pass
+        return raw, {'value': value}
+    return raw, {}
+
+
+@functools.lru_cache()
+def import_func(func_path: Union[str, Callable]) -> Callable:
+    """Return function imported from a fully qualified name."""
+    if callable(func_path):
+        return func_path
+    try:
+        module_name, func_name = func_path.rsplit(".", maxsplit=1)
+        module = importlib.import_module(module_name)
+        func = getattr(module, func_name)
+    except (ValueError, ImportError, AttributeError) as exc:
+        msg = f"Cannot import callback {func_path!r} ({exc})"
+        raise ValueError(msg) from exc
+    assert callable(func), f"{func_path} is not callable"
+    return func
+
+
+default_handlers = [
+    raw_infer_vr_handler,
+    raw_convert_exception_handler,
+    raw_LUT_descriptor_handler
+]
+
+
+def DataElement_from_raw(
+    raw_data_element: RawDataElement,
+    encoding: Optional[Union[str, MutableSequence[str]]] = None,
+    dataset: Optional["Dataset"] = None
+) -> DataElement:
+    """Return a :class:`DataElement` created from `raw_data_element`.
+
+    Call the configured data_element_callbacks to do relevant
+    pre/post-processing and convert values from raw to native types
+
+    Parameters
+    ----------
+    raw_data_element : RawDataElement
+        The raw data to convert to a :class:`DataElement`.
+    encoding : str or list of str, optional
+        The character encoding of the raw data.
+    dataset : Dataset, optional
+        If given, used to resolve the VR for known private tags.
+
+    Returns
+    -------
+    DataElement
 
+    """
+    raw = raw_data_element
+    # Legacy callbacks if set
+    callback_kwargs: dict[str, Any] = {}
+    if config.data_element_callback:
+        raw = config.data_element_callback(
+            raw_data_element,
+            encoding=encoding,
+            **config.data_element_callback_kwargs
+        )
+        # Use default handlers if legacy callbacks are set
+        cbs = default_handlers
+    else:
+        # Otherwise use those provided in settings
+        cbs = config.settings.data_element_callbacks
+        callback_kwargs = config.settings.data_element_callback_kwargs
+    for cb_path in cbs:
+        # Allow for string path or direct callable
+        cb = import_func(cb_path)
+        raw, out_kwargs = cb(
+            raw,
+            encoding=encoding,
+            dataset=dataset,
+            **callback_kwargs
+        )
+        # Update kwargs for next callback.
+        callback_kwargs.update(out_kwargs)
+    # Allow callbacks to override raw.VR
```

Comment:
Maybe I haven't followed this through properly, but returning the value in a dict to be added to callback_kwargs seems a little awkward.  For the use as shown, just returning `value` directly would seem simpler.  But perhaps I am missing a more general use case?

Reply:
So the reason I did it like this is that some callbacks, in particular the `LUT_descriptor_handler` needs to operate on the `converted` value, as a post-processing sort of step, so just passing in the `raw_data_element` wouldn't be enough for it, since we don't want to update the `RawDataElement.value` field.

I could make each handler use a signature like `(raw: RawDataElement, value: Any = None, **kwargs) -> (RawDataElement, value: Any)`, but I think that would look nearly the same. 

I could also see use cases where someone want's to set VR in the `callback_kwargs` dict and use that for the final conversion step, so both `value` and `VR` could be set by the handlers but _not_ updated on the `RawDataElement` itself.

Happy to revisit this if it's too awkward or I'm mising something.